cli/ops/native_plugins.rs

```diff
@@ -0,0 +1,103 @@
+use super::dispatch_json::{Deserialize, JsonOp, Value};
+use crate::fs as deno_fs;
+use crate::ops::json_op;
+use crate::state::ThreadSafeState;
+use deno::*;
+use dlopen::symbor::Library;
+use std::collections::HashMap;
+use std::ffi::OsStr;
+use std::sync::Arc;
+use std::sync::Mutex;
+
+pub fn init(
+  i: &mut Isolate,
+  s: &ThreadSafeState,
+  isolate_arc: Arc<Mutex<deno::Isolate>>,
+) {
+  let isolate_arc_ = isolate_arc.clone();
+  i.register_op(
+    "open_native_plugin",
+    s.core_op(json_op(s.stateful_op(move |state, args, zero_copy| {
+      op_open_native_plugin(&isolate_arc_, state, args, zero_copy)
+    }))),
+  );
+}
+
+fn open_plugin<P: AsRef<OsStr>>(lib_path: P) -> Result<Library, ErrBox> {
+  debug!("Loading Native Plugin: {:#?}", lib_path.as_ref());
+
+  Library::open(lib_path).map_err(ErrBox::from)
+}
+
+struct NativePluginResource {
+  lib: Library,
+  ops: Vec<(String, OpId)>,
+}
+
+impl Resource for NativePluginResource {}
+
+struct InitContext {
+  isolate: Arc<Mutex<deno::Isolate>>,
+  plugin_rid: ResourceId,
+  ops: Vec<(String, OpId)>,
+}
+
+impl PluginInitContext for InitContext {
+  fn register_op(
+    &mut self,
+    name: &str,
+    op: Box<dyn Fn(&[u8], Option<PinnedBuf>) -> CoreOp + Send + Sync + 'static>,
+  ) -> OpId {
+    let mut i = self.isolate.lock().unwrap();
```

Comment:
Regarding my previous comment; it seems possible to make `InitContext` get only hashmap of names and ops, I believe it's better than exposing isolate there :+1: